ludwig/datasets/loaders/hugging_face.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) 2022 Predibase, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import logging
+
+import datasets
+import pandas as pd
+
+from ludwig.datasets.loaders.dataset_loader import DatasetLoader
+
+TRAIN = "train"
+VALIDATION = "validation"
+TEST = "test"
+splits = [TRAIN, VALIDATION, TEST]
+
+
+class HFLoader(DatasetLoader):
+    def load_hf_to_dict(self, hf_id: str, hf_subsample: str) -> dict:
+        # Convert from HF DatasetDict type to dict
+        dataset_dict = datasets.load_dataset(path=hf_id, name=hf_subsample)
+        new_dict = {}
+        for split in dataset_dict:
+            new_dict[split] = dataset_dict[split].to_pandas()
+        return new_dict
+
+    def load(self, split=False, kaggle_username=None, kaggle_key=None) -> pd.DataFrame:
```

Comment:
Since we're already overriding this method, let's remove the kaggle credentials, and pass in the `huggingface_dataset_id` and the `huggingface_subsample` as explicit arguments instead of requiring the the caller set stateful variables implicitly with `setattr`.